Jetpack: adds Backup screen

leandroalonso opened the pull request:
Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/15193

This PR adds the Backup screen under the Jetpack section.

<img src="https://user-images.githubusercontent.com/7040243/103813355-bd5b3e00-503e-11eb-8d8f-192cf60b678e.gif" width="300">

@momo-ozawa I'm using the same feature flag as backup/rewind. Let me know if you think this is a problem. :) 

### To test

1. Open any site detail
2. Tap Backup
3. You should see a list containing ONLY items that can be downloaded or rewinded
4. Scroll down and compare your list to the one in wordpress.com

Filtering by date:

1. Open any site detail
2. Tap Backup
3. Select a single date or a date range
4. You should see only items that can be downloaded or rewinded in the range or day you selected

This PR only covers showing the Backup screen. This will be covered in future PRs:

* Checking if the site has the Backup feature
* Changing the error messages
* Refactoring `ActivityListViewController` and `BackupListViewController` to make the inheritance more obvious
* Updating the events

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

leandroalonso committed:
Update icon and move the backup option up

leandroalonso committed:
Adds a Backup List View Controller

leandroalonso committed:
Adds an option in the ViewModel to handle only rewindable items

leandroalonso committed:
SHow the backup screen

leandroalonso committed:
Simplify the Backup initialization

momo-ozawa commented on WordPress/Classes/Stores/ActivityStore.swift:
```diff
@@ -72,6 +72,8 @@ class ActivityStore: QueryStore<ActivityStoreState, ActivityQuery> {
     private let activityServiceRemote: ActivityServiceRemote?
 
+    var onlyRewindableItems = false
+
```
I wonder if we should call this `onlyRestorableItems` instead to keep in line with the `rewind` -> `restore` rebrand?

leandroalonso replied:
Agree, good catch! Changed in 21cc34bbef8b495a06abe72f18f36b73e9428a89

momo-ozawa commented on WordPress/Classes/Stores/ActivityStore.swift:
```diff
@@ -236,9 +238,16 @@ private extension ActivityStore {
+                actionDispatcher.dispatch(ActivityAction.receiveActivities(
+                                            site: site,
+                                            activities: self.onlyRewindableItems ? activities.filter { $0.isRewindable } : activities,
+                                            hasMore: hasMore, loadingMore: loadingMore))
```
Alignment for this method seems to be off 🙏

leandroalonso replied:
Done in a6345392498dcc192b2bc99a45d41356bf61a15b

momo-ozawa commented:
@leandroalonso I noticed that we should be using the site's timezone for the date filter calendar. It seems like the date filter calendar is set to use your device timezone.
  
I'm testing a site where the site's timezone doesn't match my device timezone, so the date date filtering results come back differently on the app vs WordPress.com.

leandroalonso committed:
Fix unit tests

Even though the added logic changes the offset, this shouldn't affect the functionality.

leandroalonso committed:
Rename onlyRewindableItems to onlyRestorableItems

leandroalonso committed:
Fix method alignment

leandroalonso commented:
@momo-ozawa I tested a site here using your timezone (Asia/Tokyo), both in the app and in WordPress.com. I selected 1-7 Jan and yet there is a Dec 31 in the results in both platforms.

I did an investigation and the issue I'm finding is that I'm not able to correctly retrieve the TimeZone for the blog, thus there's no way to adjust the date formatting:

```
(lldb) po blog.settings?.timezoneString
▿ Optional<String>
  - some : ""
```

In order to fix that I wonder that perhaps we should call service and then update this value... which is a little bit complex 😬 (not to take into account that TimeZones are quite a rabbit hole).

Plus, I'm seeing this banner on the web version:

<img src="https://user-images.githubusercontent.com/7040243/103927386-0b347c80-50f9-11eb-9f5f-376e0eae28dc.png">

I was talking to @emilylaguna that perhaps we can leave that outside of this MVP. The assumption is that not that many users will be in such different timezones. What do you think?

Also, please let me know if you have any idea on how to fix that with an easier approach.

leandroalonso commented:
@momo-ozawa Just an update on the above, the timezone was empty because no timezone was specified. 😅

But even with this change, I'm still seeing 31 Dec results (on the web and in the app — plus, applying the site's timezone in the app logic):

<img src="https://user-images.githubusercontent.com/7040243/103929331-edb4e200-50fb-11eb-8234-74575a70a2ac.png">

leandroalonso committed:
Hide the activity type filter

momo-ozawa reviewed (approved):
@leandroalonso Works as described!

Thanks for investigating the timezone issue!!! 🙏 
Strange that Dec 31 shows up in results, but I don't think it's a blocker.
